microsoft/testsuites/cpu/cpusuite.py

```diff
@@ -51,3 +61,69 @@ class CPUSuite(TestSuite):
     )
     def verify_cpu_hot_plug(self, log: Logger, node: Node) -> None:
         verify_cpu_hot_plug(log, node)
+
+    @TestCaseMetadata(
+        description="""
+            This test will check cpu hotplug with storage workload.
+            The cpu hotplug steps are same as `verify_cpu_hot_plug` test case.
+            """,
+        priority=3,
+        requirement=simple_requirement(
+            min_core_count=32,
+        ),
+    )
+    def verify_cpu_offline_storage_workload(self, log: Logger, node: Node) -> None:
+        # run fio process asynchronously on the node and give it 10 seconds to start
+        fio_storage_workload_script = CustomScriptBuilder(
+            Path(__file__).parent.joinpath("scripts"), ["fio_storage_worload.sh"]
+        )
+        process = node.tools[Fio].run_script_async(fio_storage_workload_script)
+        time.sleep(10)
+
+        # verify cpu hotplug functionality
+        verify_cpu_hot_plug(log, node)
+
+        # verify that the fio was running when hotplug was triggered
+        assert_that(
+            process.is_running(), "Storage workload was not running during CPUhotplug"
+        ).is_true()
+
+    @TestCaseMetadata(
+        description="""
+            This test will check cpu hotplug with network workload.
+            The cpu hotplug steps are same as `verify_cpu_hot_plug` test case.
+            """,
+        priority=3,
+        requirement=simple_requirement(
+            min_core_count=32,
+            min_count=2,
+        ),
+    )
+    def verify_cpu_offline_network_workload(
+        self, log: Logger, environment: Environment
+    ) -> None:
+        server = cast(RemoteNode, environment.nodes[0])
+        client = cast(RemoteNode, environment.nodes[1])
+
+        # run iperf3 process asynchronously on the server and client,
+        # and give it 10 seconds to start
+        # server.tools[Iperf3].run_as_server_async()
+        fio_network_client_workload_script = CustomScriptBuilder(
+            Path(__file__).parent.joinpath("scripts"),
+            ["fio_network_client_workload.sh"],
+        )
+        client_iperf3_process = client.tools[Iperf3].run_script_async(
```

Comment:
use iperf3 tool

Reply:
`iperf3` needs to be run in a loop asynchronously. Currently in Lisa, I don't think there is a way to run this entire script with loop asynchronously. We can only run the statement "iperf3 -c $targetIPAddress -t 60 -P 32 >> workload.json" asynchronously

Reply:
we can launch iperf3 from the tool asynchronously, if not, please add it into the tool.